logger.go

```diff
@@ -79,3 +78,45 @@ func Sublogger(inp ZapCompatibleLogger, subname string) (loggerRet ZapCompatible
 
 	return loggerRet
 }
+
+// LogWith adds fields for logging to a given ZapCompatibleLogger instance.
+// This function uses reflection to dynamically add fields to the provided logger by
+// calling its `WithFields` method if it is an RDK logger, or its `With` method if it is a Zap logger.
+// If neither method is available, it logs a debug message and returns the original logger.
+func LogWith(inp ZapCompatibleLogger, args ...interface{}) (loggerRet ZapCompatibleLogger) {
+	loggerRet = inp
+
+	defer func() {
+		if r := recover(); r != nil {
+			inp.Debugf("panic occurred while adding fields to logger: %v, returning self", r)
```

Comment:
The use of a named return value for defer+recover is perfectly idiomatic, but it's not used super often.

So I think a comment here saying this case will return `loggerRet` that was initialized to `inp` would help a reader. And maybe adding it's intentionally never re-assigned before calling functions can panic would be good.